src/Bicep.LangServer/Telemetry/TelemetryProvider.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+
+namespace Bicep.LanguageServer.Telemetry
+{
+    public class TelemetryProvider : ITelemetryProvider
+    {
+        public ILanguageServer? LanguageServer { get; set; }
```

Comment:
Use constructor injection and keep the class members immutable. See e.g. `BicepCompilationManager` for an example of this.

Reply:
Updated as part of this commit:
https://github.com/Azure/bicep/commit/ffbef4cf981f55f6966f50de6ed4dec09bd90905